### PR TITLE
refs #001: Kellett milestone 1 migrations

### PR DESCRIPTION
**Pull request by dpfitzsi**
# Summary

** Update from #19 to test build process with different branch name.

This pull request adds migrations for home page, landing page, landing page level 2. This includes process plugins to locate the author and process paragraphs. 

Configuration for content types and associated views will be added as a separate pull request.

# Known Issues

There are no known bugs or issues with code at this time.

# Rollout Instructions

1. Rollout code.
2. Run taxonomy migrations.
3. Import homepage, landing_page, and landing_page_level_2 content and translations.
e.g. `drush migrate:import --tag="Landing Page Level 2"`.

### CODE REVIEW

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+migration_group: legacy
+source:
+    plugin: d7_node_entity_translation
+    node_type: homepage
+    constants:
+        default_text_format: full_html
+destination:
+    plugin: entity:node
+    translations: true
+    destination_module: content_translation
```
Comment (ew-andrew-sus) [inline docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml:17]: add bundle here
default_bundle: homepage

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+process:
+    uid:
+        plugin: yukon_author
+    nid:
+        plugin: migration_lookup
+        migration: yukon_migrate_home_page
+        source: entity_id
+        no_stub: true
+    type:
```
Comment (ew-andrew-sus) [inline docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml:26]: would be better to add type in destination as default_bundle

**Review by ew-andrew-sus — COMMENTED** [review]
Do not use custom plugins to migrate paragraph/media/users please
There are already some migrations ready:
Media: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/media
Paragraphs: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/paragraphs

You can reuse or create similar ones.

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    method: row
+    not_equals: true
+    value: 1
+    source: status
+  created: created
+  changed: timestamp
+  promote: promote
+  sticky: sticky
+
+  metatags/0/description: field_page_description/0/value
```
Comment (ew-andrew-sus) [inline docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page.yml:33]: Create a separate field as it was in D7 and migrate value into it. Matatags should be configured to use tat field token. So do not migrate anything directly into matatags please

Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page.yml]: This should now be resolved.

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml (hunk 1 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml]: This is resolved.

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml (hunk 2 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml:25]: This is resolved.

**Comment by dpfitzsi** [comment]
> Do not use custom plugins to migrate paragraph/media/users please There are already some migrations ready: Media: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/media Paragraphs: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/paragraphs
> 
> You can reuse or create similar ones.

I have changed the migration over to yml and added a new source plugin for the one media type (icon).

File: docroot/modules/custom/yukon_migrate/src/Plugin/migrate/source/YukonFilemime.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * The 'yukon_migrate_file_mime' source plugin.
+ *
+ * @MigrateSource(
+ *   id = "yukon_migrate_file_mime",
+ *   source_module = "file"
+ * )
+ */
+class YukonFilemime extends File {
```
Comment (alt-ew) [inline docroot/modules/custom/yukon_migrate/src/Plugin/migrate/source/YukonFilemime.php:15]: We already have a custom source plugin for files - YukonFile. There is no point to create a new one to add this simple logic. Can you please move this source plugin logic to the YukonFile source plugin?

File: docroot/modules/custom/yukon_migrate/migrations/paragraphs/yukon_migrate_secondary_content.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  plugin: d7_paragraphs_item
+  bundle: secondary_content
+  constants:
+    default_text_format: full_html
+process:
+
+    field_category_title: field_category_title
+    field_subcategory_links:
+        -   plugin: sub_process
+            source: field_subcategory_links
```
Comment (alt-ew) [inline docroot/modules/custom/yukon_migrate/migrations/paragraphs/yukon_migrate_secondary_content.yml:18]: According to data from the database, this field contains links like 'node/$nid'. During the migration process, we don't keep node ids, therefore we need to replace old node URLs with the new correct ones. For this purpose, we have a new process plugin which is called - Links. Could you please re-work this section to make it use this plugin?

**Review by alt-ew — CHANGES_REQUESTED** [review]
Looks like this PR contains changes from other branches.
@dpfitzsi could you please merge the master branch to this branch? Maybe it will resolve the problem.

File: docroot/modules/custom/yukon_migrate/src/Plugin/migrate/source/YukonFilemime.php (hunk 4 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/src/Plugin/migrate/source/YukonFilemime.php]: This is now resolved.

File: docroot/modules/custom/yukon_migrate/migrations/paragraphs/yukon_migrate_secondary_content.yml (hunk 5 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/paragraphs/yukon_migrate_secondary_content.yml]: This is now resolved.

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    sticky: sticky
+    revision_log: log
+    field_page_description/value: field_page_description/0/value
+    field_page_description/format: constants/default_text_format
+    body/value: field_opener_description/0/value
+    body/format: constants/default_text_format
+    field_icon_dark:
+        plugin: migration_lookup
+        source: field_svg_upload/0/fid
+        migration: yukon_migrate_icon
```
Comment (alt-ew) [inline docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2.yml:42]: Do we need to add all of the migrations that are used in the migration lookup process to the migration dependencies?

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2_translations.yml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        plugin: migration_lookup
+        migration: yukon_migrate_teams
+        source: field_yukon_editorial_team
+        no_stub: true
+    field_social_sharing: field_social_sharing
+    content_translation_source: source
+    content_translation_outdated: translate
+migration_dependencies:
+    required:
+        - yukon_migrate_landing_page_level_2
```
Comment (alt-ew) [inline docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2_translations.yml:83]: Can you please update the migration dependencies list and add here all required and optional migrations?
For example, your migration has the migration lookup in the yukon_migrate_teams migration, which means that this migration should be a dependency.
Could you please go through all your migrations and fix this? :)

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2.yml (hunk 6 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2.yml:42]: This has been resolved

File: docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2_translations.yml (hunk 7 above)
Comment (dpfitzsi) [reply docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2_translations.yml:83]: This has been resolved

**Review by alt-ew — APPROVED** [review]

**Pull request merged**